File: 1401-1500/1403-minimum-subsequence-in-non-increasing-order.py
class Solution(object):
    def minSubsequence(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        su = sum(nums)
        idx = list(range(len(nums)))
        idx.sort(key=lambda i: nums[i], reverse=True)
        se_su = 0
        se_idx = []
        for i in idx:
            se_idx.append(i)
            se_su += nums[i]
            if se_su > su - se_su:
                break
        return [nums[i] for i in se_idx]


if __name__ == '__main__':
    a = [4, 3, 10, 9, 8]
    print(Solution().minSubsequence(a))

File: 1401-1500/1403-minimum-subsequence-in-non-increasing-order.cpp
#include "common.h"

class Solution {
public:
    vector<int> minSubsequence(vector<int>& nums) {
      sort(nums.begin(), nums.end(), greater<int>());
      int su = accumulate(nums.begin(), nums.end(), 0);
      int half = su / 2;
      int h = 0;
      for (int i = 0; i < nums.size(); ++i) {
        h += nums[i];
        if (h > half) {
          nums.resize(i + 1);
          return nums;
        }
      }
      return nums;
    }
};

int main() {
  vector<int> nums{4, 3, 10, 9, 8};
  PRINT_ARRAY(Solution().minSubsequence(nums));
  return 0;
}
